.github/workflows/code-coverage.yml

```diff
@@ -0,0 +1,57 @@
+name: code-coverage
+
+on:
+  push:
+    branches: [master]
+
+env:
+  java: 8
+  distribution: "adopt"
+
+jobs:
+  code-coverage:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK (${{ env.java }} ${{ env.distribution }})
+        uses: actions/setup-java@v2
+        with:
+          distribution: ${{ env.distribution }}
+          java-version: ${{ env.java }}
+
+      - name: Cache Maven Packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: Build with Maven
+        run: mvn -B test
+
+      - name: Generate JaCoCo Badge
+        id: jacoco
+        uses: cicirello/jacoco-badge-generator@v2.0.1
+
+      - name: Log coverage percentage
+        run: |
+          echo "coverage = ${{ steps.jacoco.outputs.coverage }}"
+          echo "branch coverage = ${{ steps.jacoco.outputs.branches }}"
+
+      - name: Commit the badge (if it changed)
+        run: |
+          if [[ `git status --porcelain` ]]; then
```

Comment:
This won't work. You'll need to use that git add and commit action that I found or use the environment variables available in github actions to build the remote url.

Reply:
This is all happening in the context of the github repo - not sure why you need to worry about the remote url? This code is as per https://github.com/marketplace/actions/jacoco-badge-generator "Example Workflow 1: Generate instructions (or C0) coverage badge only."

Reply:
https://github.com/Flagsmith/flagsmith-java-client/runs/2602360892?check_suite_focus=true

Reply:
Pretty sure this is due to two concurrent builds running then failing as they are both trying to commit to master and getting out of sync. Hence moving the action out into its own workflow that only runs by itself.

Reply:
The error message certainly doesn't look like that's the case. But if you can prove that this works when the coverage changes then fine.

Reply:
Deleted the badge and had it recommitted here https://github.com/Flagsmith/flagsmith-java-client/commit/ea4d6cadc06ea57963888ae03fc5a018d0390997  
https://github.com/Flagsmith/flagsmith-java-client/runs/2611326531?check_suite_focus=true